client/src/modules/list/components/Items/ListItemName/ListItemName.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import { updateListItem } from '../model/actions';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+import { KeyCodes } from 'common/constants/enums';
+import { stopPropagation } from 'common/utils/helpers';
+import Preloader from 'common/components/Preloader';
+
+class ListItemName extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    const { name } = this.props;
+
+    this.state = {
+      isNameInputFocused: false,
+      isTipVisible: false,
+      name,
+      pending: false
+    };
+
+    this.nameInput = React.createRef();
+    this.listItemName = React.createRef();
+  }
+
+  componentDidUpdate() {
+    const { isNameInputFocused, name } = this.state;
+
+    if (isNameInputFocused) {
+      document.addEventListener('keydown', this.handleKeyPress);
+      document.addEventListener('mousedown', this.handleClickOutside);
+    }
+
+    if (!isNameInputFocused) {
+      document.removeEventListener('keydown', this.handleKeyPress);
+      document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+
+    if (name.length === 0) {
+      this.nameInput.current.focus();
+    }
+  }
+
+  handleKeyPress = event => {
+    const { code } = event;
+
+    if (code === KeyCodes.ENTER || code === KeyCodes.ESCAPE) {
+      this.handleNameUpdate();
+    }
+  };
+
+  handleNameUpdate = () => {
+    const { name: updatedName } = this.state;
+    const {
+      itemId,
+      match: {
+        params: { id: listId }
+      },
+      name,
+      updateListItem
+    } = this.props;
+    const isNameUpdated = updatedName !== name;
+    const canBeUpdated = updatedName.trim().length > 1;
+
+    if (canBeUpdated && isNameUpdated) {
+      this.setState({ pending: true });
+
+      updateListItem(listId, itemId, { name: updatedName }).then(() => {
+        this.setState({
+          pending: false,
```

Comment:
As we discussed: because setting pending depends on async action when the component is unmounted setting state causes an error...